### ISSUE TITLE
LibCURL fails to build from source when BUILD_MACHINE=arm64-apple-darwin20.2.0

### ISSUE THREAD

**Issue by omus** [JuliaLang/julia#39627]
When attempting to build LibCURL from source on Apple Silicon the following error is encountered.
```
checking curl version... 7.73.0
checking build system type... Invalid configuration `arm64-apple-darwin20.2.0': machine `arm64-apple' not recognized
configure: error: /bin/sh /Users/omus/Development/Julia/arm64/latest-src/deps/srccache/curl-7.73.0/config.sub arm64-apple-darwin20.2.0 failed
make: *** [/Users/omus/Development/Julia/arm64/latest-src/deps/curl.mk:50: scratch/curl-7.73.0/build-configured] Error 1
make: Leaving directory '/Users/omus/Development/Julia/arm64/latest-src/deps'
```
Modifying the `BUILD_MACHINE` to be `aarch64-apple-darwin20.2.0` does allow LibCURL to build but changing this globally may not work for all external dependencies. There are a few other Apple Silicon build issues to work through before this can be validated.

Part of: https://github.com/JuliaLang/julia/issues/36617

**Comment by omus** [comment]
cc: @staticfloat

**Comment by Keno** [comment]
We vendor a copy of the autotools utilities for just this situation. The libcurl build probably needs to be updated to copy those over.

**Comment by giordano** [comment]
For example 
https://github.com/JuliaLang/julia/blob/50742c7e42b8d55fdb01976c18bd237ad8e72787/deps/mpfr.mk#L34

**Comment by omus** [comment]
> We vendor a copy of the autotools utilities for just this situation. The libcurl build probably needs to be updated to copy those over.

Ah, thanks. Seems obvious in retrospect

**Comment by omus** [comment]
Included the fix in #39625
